### PR TITLE
CLN: core.internals

### PR DESCRIPTION
**Pull request by jbrockmendel**
some annotations, remove an unused argument, privatize where possible, avoid a runtime/circular import

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -3004,9 +3003,8 @@ def _extend_blocks(result, blocks=None):
                 blocks.extend(r)
             else:
                 blocks.append(r)
-    elif isinstance(result, BlockManager):
-        blocks.extend(result.blocks)
     else:
+        assert isinstance(result, Block), type(result)
```
Comment (simonjayhawkins) [inline pandas/core/internals/blocks.py]: do we need a run-time check? can result be typed as Union[Block, List[Union[Block, List[Block]]]? Might help to type blocks and return type while touching this.

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: thats a pretty ugly nested union; ill see if i can get rid of the double-list in the next pass

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: @simonjayhawkins is this a deal-breaker?  id like to clear the queue

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**